test/spoof/spoof.go

```diff
@@ -174,3 +189,33 @@ func (sc *SpoofingClient) Poll(req *http.Request, inState ResponseChecker) (*Res
 
 	return resp, err
 }
+
+// LogZipkinTrace provides support to log Zipkin Trace for param: traceID
+func (sc *SpoofingClient) LogZipkinTrace(traceID string) error {
+
+	if zipkin.CheckZipkinPortAvailability() {
+		return errors.New("Zipkin Port(9411) is not-setup. failing Zipkin Trace retrieval")
+	}
+
+	sc.logger.Infof("Logging Zipkin Trace: %s", traceID)
+
+	zipkinTraceEndpoint := fmt.Sprintf(zipkin.ZipkinTraceEndpoint, traceID)
+
+	// Sleep to ensure all traces are correctly pushed on the backend.
+	time.Sleep(5 * time.Second)
+	resp, err := http.Get(zipkinTraceEndpoint)
+
+	if err != nil {
+		return errors.New("Error retrieving Zipkin trace")
+	}
+
+	trace, err := ioutil.ReadAll(resp.Body)
+
+	if err != nil {
+		errors.New("Error reading Zipkin trace response")
+	}
+
+	sc.logger.Infof(string(trace))
```

Comment:
What's the expected log output here? Wouldn't be better formatting it somehow?

Reply:
The trace log itself is a JSON payload representing different traces. Not sure if you thinking about pretty printing of JSON otherwise there is not much we can format.

Reply:
I'm concerned that this would print a blob hard to read, thus useless. Pretty printing is a possible solution, but it depends on the objective of this log entry: is it intended to be read by a human?

Reply:
Yes, it would be read by human. I have pretty printed the JSON output.

Reply:
Nice. Can you please add an example of the output as a comment to this PR?

Reply:
Edit your comment and add the proper markdown formatting (3 backticks before and after the block) and it should be rendered correctly.

Reply:
Is that a single log entry? I was expecting something way smaller. I'm concerned that this will flood the test logs. Can you log just the relevant fields (or assemble, say, an inspection URL with the relevant fields)?

Reply:
I recommend dumping the whole trace since I'm not sure what part we'll need.  But only do it once.  E.g. if we get 10 502's, just dump the trace for the first one and then log how many 502's we received in total.

Reply:
The call to LogZipkinTrace() would be made from the test, so the logic to aggregate the number of 502s (or other condition depending on the test)would reside within the test.

Reply:
Based on the comments above, leaving the current logging statement as is.